src/relay/backend/interpreter.cc

```diff
@@ -774,6 +751,14 @@ class Interpreter :
   // Cache ops that need to be frequently used later to reduce lookup overhead.
   const Op& debug_op_;
   const Op& shape_of_op_;
+  // The free vars captured by the last closure.
+  std::unordered_map<Closure, std::vector<Var>, ObjectHash, ObjectEqual> closure_captured_vars_;
```

Comment:
One drawback using `closure_captured_vars_` is that the memory leak as it will never be released. I wonder if we should create a separate closure object for interpreter that inherits from `ClosureObj` and stores the captured var expr.

Reply:
wait... what?
closure got it's name because closure store the free variables it need.
Is there any reason for this design? This is just moving the member from inside the closure to a map.

Reply:
@MarisaKirisame Are you saying why I used the map here? I was trying to leverage `VMClousre` which saves the evaluated values of the captured free vars.

@icemelon9 we can probably have a destructor to clean the allocated closures in the map. Another way is probably just to have a base `ClosureObj` class and let `VMClosureObj` and `InterpreterClosureObj` inherit from it so most of original code here keep unchanged. How do you think?

Reply:
I'm good with the design after the change. @MarisaKirisame Could you take another look?